From a362698561bdb269a071fa0571e147c56232076c Mon Sep 17 00:00:00 2001
From: Brock Noland <brock@apache.org>
Date: Thu, 19 Feb 2015 09:13:00 -0800
Subject: [PATCH 0035/1164] CLOUDERA-BUILD: CDH-25351 commit for
 Hadoop23Shim.isLocalMode is incorrect (was: hs2 +
 beeline: select count(*) from a tables,
 encountered java.lang.NullPointerException from
 mr jobs)

---
 .../apache/hadoop/hive/shims/Hadoop23Shims.java    |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
index b800266..9b2c8ba 100644
--- a/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
+++ b/shims/0.23/src/main/java/org/apache/hadoop/hive/shims/Hadoop23Shims.java
@@ -194,8 +194,10 @@ public TaskAttemptID newTaskAttemptID(JobID jobId, boolean isMap, int taskId, in
 
   @Override
   public boolean isLocalMode(Configuration conf) {
-    return "local".equals(conf.get("mapreduce.framework.name")) ||
-      "local".equals(conf.get("mapred.job.tracker"));
+    if (isMR2()) {
+      return "local".equals(conf.get("mapreduce.framework.name"));
+    }
+    return "local".equals(conf.get("mapred.job.tracker"));
   }
 
   @Override
-- 
1.7.9.5

